salttesting/jenkins.py

```diff
@@ -941,12 +1044,222 @@ def delete_parallels_vm(options):
         return 1
 
 
+def check_win_minion_connected(options):
+
+    if 'salt_minion_bootstrapped' not in options:
+        print_bulleted(
+            options, 'Minion not bootstrapped. Not pinging minion.', 'RED')
+        sys.exit(1)
+
+    # Check to see if the minion was rebooted after salt install
+    # This is needed for c:\salt to be found in the path
+    # salt-call will not work it it's not in the path
+    # Because the salt service is running as System, it gets a different
+    # environment. The only way to refresh the environment System uses is to
+    # reboot. We probably need to add a reboot option to the Salt installer
+
+    # Check this option in case this is run more than once, we don't want to
+    # reboot more than needed
+    if not getattr(options, 'salt_minion_rebooted', False):
+
+        # Make sure the minion is connected by returning a ping, then reboot
+        print_bulleted(options, 'Pinging bootstrapped minion ... ')
+        cmd = ['salt', '--out=json', '-l', options.log_level,
+               options.vm_name, 'test.ping']
+
+        # Attempt to connect to the new minion, it can take a while with a new
+        # install
+        retries = 0
+        while retries <= 12:
+
+            retries += 1
+            stdout, stderr, exitcode = run_command(
+                cmd, options, return_output=True, stream_stdout=False,
+                stream_stderr=False)
+            if exitcode:
+                print_bulleted(
+                    options,
+                    'Failed to return a ping from the minion. Exit code: {0}'
+                    ''.format(exitcode), 'RED'
+                )
+                if retries >= 12:
```

Comment:
I think you may have an off-by-one error here. Since you increment above, when you start the 13th loop, you'll have the counter sent to 12 here, which will exit and this is probably not what you want.